modules/docs/arrow-docs/docs/docs/generic/coproduct/README.md

```diff
@@ -0,0 +1,167 @@
+```
+
+##### fold
+
+Obviously, we're not just modeling errors for fun, we're going to handle them! All Coproducts have `fold` which allows us to condense the Coproduct down to a single type. For example, we could handle errors as such in a UI:
+
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.Coproduct3
+import arrow.generic.coproduct3.fold
+
+fun handleCommonError(commonError: CommonServerError) {
+}
+
+fun showCarAlreadyRegistered() {
+}
+
+fun callPolice() {
+}
+
+fun showCarSuccessfullyRegistered(car: Car) {
+}
+
+fun renderApiResult(apiResult: Coproduct3<CommonServerError, RegistrationError, Registration>) = apiResult.fold(
+            { commonError -> handleCommonError(commonError) },
+            { registrationError ->
+                when (registrationError) {
+                    CarAlreadyRegistered -> showCarAlreadyRegistered()
+                    StolenCar -> callPolice()
+                }
+            },
+            { registration ->
+                val registeredCar = registration.car
+
+                showCarSuccessfullyRegistered(registeredCar)
+            }
+    )
+```
+
+This example returns `Unit` because all of these are side effects, let's say our application was built for a command line and we just have to show a `String` for the result of the call (if only it was always that easy):
+
+```kotlin:ank
+import arrow.generic.*
+import arrow.generic.coproduct3.Coproduct3
+import arrow.generic.coproduct3.fold
+
+fun renderApiResult(apiResult: Coproduct3<CommonServerError, RegistrationError, Registration>): String = apiResult.fold(
+            { commonError ->
+                when (commonError) {
+                    ServerError -> "Server error, try again later."
+                    UserUnauthorized -> "Unauthorized!"
+                    OverRequestLimit -> "Too many api requests, try again later."
+                }
+            },
+            { registrationError ->
+                when (registrationError) {
+                    CarAlreadyRegistered -> "Car already registered."
+                    StolenCar -> "Car reported stolen!"
+                }
+            },
+            { registration ->
+                "Successfully Registered: $registration.car"
+            }
+    )
+```
+
+Here we're able to return the result of the `fold` and we're forced to handle all cases! Neat!
+
+##### select
+
+Let's say we also want to store the `Registration` object into our database when we successfully register a car. We don't really want to have to `fold` over every single case just to handle something for the `Registration`, this is where `select<T>` comes to the rescue! We're able to take a Coproduct and `select` the type we care about from it. `select` returns an `Option`, if the value of the Coproduct was for the type you're trying to `select`, you'll get `Some`, if it was not the type used with `select`, you'll get `None`.
```

Comment:
Maybe we could move all the first part until "...select comes to the rescue!" to the previous section. Then you present select as a new section, which would start on "We're able to take a...". That helps to start introducing the concept by the need for it before even giving it a name. It's just a more gradual way to put it.